APIComparer.Core/ComparerEngine.cs

```diff
@@ -22,30 +24,49 @@ public Diff CreateDiff(string leftAssembly, string rightAssembly)
         {
             return CreateDiff(ReadTypes(leftAssembly,true), ReadTypes(rightAssembly,true));
         }
+
         public Diff CreateDiff(AssemblyGroup leftAssemblyGroup, AssemblyGroup rightAssemblyGroup)
         {
             if (rightAssemblyGroup is EmptyAssemblyGroup)
             {
                 return new EmptyDiff();
             }
-            return CreateDiff(ReadTypes(leftAssemblyGroup.Assemblies,leftAssemblyGroup.ReadSymbols), ReadTypes(rightAssemblyGroup.Assemblies,rightAssemblyGroup.ReadSymbols));
+
+            return CreateDiff(ReadTypes(leftAssemblyGroup.Assemblies, leftAssemblyGroup.ReadSymbols), ReadTypes(rightAssemblyGroup.Assemblies, rightAssemblyGroup.ReadSymbols));
         }
-        IEnumerable<TypeDefinition> ReadTypes(IEnumerable<string> assemblyGroup,bool readSymbols)
+
+        IEnumerable<TypeDefinition> ReadTypes(IEnumerable<string> assemblyGroup, bool readSymbols)
         {
             var readerParams = new ReaderParameters
             {
                 ReadSymbols = readSymbols
             };
-            return assemblyGroup.SelectMany(assembly => AssemblyDefinition.ReadAssembly(assembly, readerParams).MainModule.Types);
+
+            return assemblyGroup
+                .Where(assembly => IsManagedAssembly(assembly))
+                .SelectMany(assembly => AssemblyDefinition.ReadAssembly(assembly, readerParams).MainModule.Types);
         }
-        IEnumerable<TypeDefinition> ReadTypes(string leftAssembly,bool readSymbols)
+
+        IEnumerable<TypeDefinition> ReadTypes(string leftAssembly, bool readSymbols)
         {
-            var readerParams = new ReaderParameters
+            return ReadTypes(new[] { leftAssembly }, readSymbols);
+        }
+
+        private bool IsManagedAssembly(string assemblyPath)
+        {
+            // From https://github.com/scriptcs/scriptcs/pull/250/files by @aaronpowell
+            // License: Apache 2.0 (https://github.com/scriptcs/scriptcs/blob/dev/LICENSE.md)
```

Comment:
/cc @aaronpowell I've ~~shamelessly stolen~~ been greatly inspired by your work in https://github.com/scriptcs/scriptcs/pull/250. Are you cool with this?